### PR TITLE
Read subobjects mapping parameter in advance

### PR DESCRIPTION
**Pull request by javanna**
As part of #86166 we added support for a new mapping parameter called `subobjects` that
can be set to the `object` field type. Such parameter will affect not only how incoming
documents will be dynamically mapped in the future, but also how field names are treated
as part of the mappings themselves.

Mappings get parsed into a map, where keys ordering is not guaranteed. In case a mappings call
contains `subobjects` set to `false` and also sub-fields for that same object, we need to make
sure that we read the parameter in advance in order to know how to treat the sub-field and decide
whether we need to expand dots in field names or leave them as they are.

### CODE REVIEW

**Review by romseygeek — APPROVED** [review]
One question but LGTM otherwise

File: server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java [hunk 1]
```diff
@@ -242,6 +240,13 @@ protected static boolean parseObjectOrDocumentTypeProperties(
             return false;
         }
 
+        protected static void parseSubobjects(Map<String, Object> node, ObjectMapper.Builder builder) {
+            Object subobjectsNode = node.remove("subobjects");
+            if (subobjectsNode != null) {
+                builder.subobjects(XContentMapValues.nodeBooleanValue(subobjectsNode, "subobjects.subobjects"));
```
[the remark this answers is not in the document]
Comment (javanna) [reply server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java:246]: no it is correct, based on my copy and paste skills

[the remark this answers is not in the document]
Comment (javanna) [reply server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java:246]: I would have asked the same though in your shoes.

**Comment by javanna** [comment]
@romseygeek I made subobjects a constructor argument in ObjectMapper.Builder , you may want to have another look when you get a chance.

**Review by romseygeek — APPROVED** [review]
LGTM, thanks @javanna

**Pull request merged**